app/controllers/links_controller.rb

```diff
@@ -5,9 +5,9 @@ class LinksController < ApplicationController
   def index
     if params[:tag]
```

Comment:
`params[:tag]` is used many times withing scope of the method. Assign it to local variable and use it.